pkg/controller/common/cleaner.go

```diff
@@ -87,23 +90,35 @@ func (c *cleaner) Start(ctx context.Context) error {
 }
 
 func (c *cleaner) Clean(ctx context.Context) error {
+	var err error
+	defer func() {
+		//placing this call inside a closure allows for result and err to be bound after tick executes
```

Comment:
are named returns a cleaner way of handling this rather than declaring the values before and always having to change a specific variable?

Named returns should make it so you don't have to always set err = something before returning. Would make it a lot harder to mess up.

If you return a value for a named return it sets that name to it so it will be used in the defer.

Reply:
If we didn't have cases where we ignore errors and just return nil - yes. We have lots of cases where we return `ctrl.Result, client.IgnoreNotFound(err)`. In that case, we'd have to call `err = client.IgnoreNotFound(err)`, which can be just as awkward.

Ideally, this would be part of a pipeline, and whether or not an error is returned will determine if the error metrics are incremented - but that's work for a different PR